src/python/pants/rules/core/run.py

```diff
@@ -1,35 +1,58 @@
 # Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+from pathlib import Path
+
+from pants.build_graph.address import Address
 from pants.engine.addressable import BuildFileAddresses
 from pants.engine.console import Console
+from pants.engine.fs import DirectoryToMaterialize, Workspace
 from pants.engine.goal import Goal
 from pants.engine.interactive_runner import InteractiveProcessRequest, InteractiveRunner
 from pants.engine.rules import console_rule
+from pants.engine.selectors import Get
+from pants.rules.core.binary import CreatedBinary
+from pants.util.contextutil import temporary_dir
 
 
 class Run(Goal):
   """Runs a runnable target."""
-  name = 'v2-run'
+  name = 'run'
 
 
+#TODO(gregs) - the `run` rule should really only accept a single target, but we don't have the infrastructure
```

Comment:
IMO, raising an exception here would be perfectly reasonable.

The upshot around `console_rule`s never depending on one another is that they are able to define their own requirements on roots, without concerning themselves with what other goals/tasks want: if the goal needs one, it can blow up if it gets `!= 1`.